packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -311,9 +311,9 @@ export const useConnectSocket: EnsembleActionHook<ConnectSocketAction> = (
   }, [
     screenData,
     socket,
-    onSocketConnectAction,
-    onMessageReceiveAction,
-    onSocketDisconnectAction,
+    onSocketConnectAction?.callback,
+    onMessageReceiveAction?.callback,
+    onSocketDisconnectAction?.callback,
```

Comment:
These changes are fine but preferably we need to memoize the action return object.